game.js

```diff
@@ -454,11 +453,24 @@ class Game {
         const winloss = (this.state.winner === this.conn.bot_id ? "W" : "L");
         this.log(`Game over.   Result: ${col}+${res}  ${winloss}`);
 
-        if (this.bot) {
-            this.bot.gameOver();
-            this.ensureBotKilled();
+        // Notify bot of end of game and send score
+        if (config.farewell_score && this.state) {
```

Comment:
Why do we need this check on `this.state`? Also we've done `this.state.winner` and a few more property reads a few lines before, so i don't think we can reach this line with `this.state` is `null` or `false`.

Reply:
this is basically  because farewell does, and i am not sure what leads to state being null

Reply:
fixed